datafusion/expr/src/udf.rs

```diff
@@ -467,7 +478,28 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
     /// to arrays, which will likely be simpler code, but be slower.
     ///
     /// [invoke_no_args]: ScalarUDFImpl::invoke_no_args
-    fn invoke(&self, _args: &[ColumnarValue]) -> Result<ColumnarValue>;
+    fn invoke(&self, _args: &[ColumnarValue]) -> Result<ColumnarValue> {
```

Comment:
Nice! -- as a follow on PR I think we should deprecate the other two functions (`invoke_no_args` and `invoke`) telling people to use `invoke` instead

Is this ok with you @jayzhan211 ?

Reply:
> as a follow on PR I think we should deprecate the other two functions (`invoke_no_args` and `invoke`) telling people to use `invoke` instead

did you mean `invoke_batch`?

yes, it would be great to have only one invoke entry-point